From 9a2ab75ea28ab55987c227c94e9b8169a977a6b9 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Tue, 12 May 2015 12:05:19 +0800
Subject: [PATCH 390/462] atomisp2:not stop css when start css failed

sometimes, start css in atomisp_recover() would failed;
then, we should not call atomisp_css_stop() when release atomisp

Change-Id: I406df844a8b362a912a2689a7c27c6a01a0280b0
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-27935
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/55531
Approver: Jing Tao <jing.tao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c       | 8 +++++---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c     | 8 ++++----
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 55596b1..be07e98 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1403,9 +1403,11 @@ static void __atomisp_css_recover(struct atomisp_device *isp, bool isp_timeout)
 					CSS_INPUT_MODE_SENSOR);
 
 		css_pipe_id = atomisp_get_css_pipe_id(asd);
-		atomisp_css_start(asd, css_pipe_id, true);
-
-		asd->streaming = ATOMISP_DEVICE_STREAMING_ENABLED;
+		if (atomisp_css_start(asd, css_pipe_id, true))
+			dev_warn(isp->dev,
+				"start SP failed, so do not set streaming to be enable!\n");
+		else
+			asd->streaming = ATOMISP_DEVICE_STREAMING_ENABLED;
 
 		atomisp_csi2_configure(asd);
 	}
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 6366b0e..be0c5ba 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -2046,10 +2046,10 @@ int __atomisp_streamoff(struct file *file, void *fh, enum v4l2_buf_type type)
 		cancel_work_sync(&asd->delayed_init_work);
 		asd->delayed_init = ATOMISP_DELAYED_INIT_NOT_QUEUED;
 	}
-
-	css_pipe_id = atomisp_get_css_pipe_id(asd);
-	ret = atomisp_css_stop(asd, css_pipe_id, false);
-
+	if (first_streamoff) {
+		css_pipe_id = atomisp_get_css_pipe_id(asd);
+		ret = atomisp_css_stop(asd, css_pipe_id, false);
+	}
 	/* cancel work queue*/
 	if (asd->video_out_capture.users) {
 		capture_pipe = &asd->video_out_capture;
-- 
1.9.1

